routers/web/admin/users.go

```diff
@@ -401,3 +401,34 @@ func DeleteUser(ctx *context.Context) {
 		"redirect": setting.AppSubURL + "/admin/users",
 	})
 }
+
+// AvatarPost response for change user's avatar request
+func AvatarPost(ctx *context.Context) {
+	u := prepareUserInfo(ctx)
+	if ctx.Written() {
+		return
+	}
+
+	form := web.GetForm(ctx).(*forms.AvatarForm)
+	if err := router_user_setting.UpdateAvatarSetting(ctx, form, u); err != nil {
+		ctx.Flash.Error(err.Error())
+	} else {
+		ctx.Flash.Success(ctx.Tr("settings.update_user_avatar_success"))
+	}
+
+	ctx.Redirect(setting.AppSubURL + "/admin/users/" + fmt.Sprint(u.ID))
```

Comment:
If the type of `u.ID` is changed `fmt.Sprint` will continue to work but would not necessarily do the correct escaping.

Use `strconv.FormatInt` instead.

```suggestion
	ctx.Redirect(setting.AppSubURL + "/admin/users/" + strconv.FormatInt(u.ID, 10))
```